dht/dht.go

```diff
@@ -141,6 +163,12 @@ func NewServer(c *ServerConfig) (s *Server, err error) {
 		}
 	}
 	s.bootstrapNodes = c.BootstrapNodes
+	if c.NodeId != "" {
+		if !c.NoSecurity {
+			return nil, ErrConflictingConfigNoSecNodeId
```

Comment:
I don't see why this is true. If security is enabled, can't we just check if their provided NodeId is secure, and only error if it isn't?